keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -327,6 +327,9 @@ class CombinedRepository(
   override fun artifactVersions(artifact: DeliveryArtifact): List<String> =
     artifactRepository.versions(artifact)
 
+  override fun artifactVersions(artifact: DeliveryArtifact, limit: Int): List<String> =
```

Comment:
Same comment on adding the `limit` parameter to the existing function instead of defining a new one.